clutel/clutel_test.go

```diff
@@ -0,0 +1,283 @@
+package clutel_test
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/alcionai/clues"
+	"github.com/alcionai/clues/clutel"
+	"github.com/alcionai/clues/internal/tester"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// TestStartSpan_Uninitialized ensures nothing panics if AddSpan is called and
+// neither clues nor OTEL is initialized.
+func TestStartSpan_Uninitialized(t *testing.T) {
+	assert.NotPanics(
+		t,
+		func() {
+			clutel.StartSpan(t.Context(), "test span")
+		},
+	)
+}
+
+// TestStartSpan_Uninitialized_Concurrent ensures that even if OTEL isn't
+// initialized there's no race condition when attempting to add spans to a
+// parent context concurrently.
+func TestStartSpan_Uninitialized_Concurrent(t *testing.T) {
```

Comment:
if the spanBuilder is used going forward, there should be a test like this for the spanBuilder way of starting spans as this is actually a difficult bug to track down

Reply:
sure, I can add that.